.jenkins/lsu/env-clang-9.sh

```diff
@@ -14,15 +14,14 @@ module load openmpi
 export HPXRUN_RUNWRAPPER=srun
 export CXX_STD="17"
 
-configure_extra_options="-DCMAKE_BUILD_TYPE=${build_type}"
-configure_extra_options+=" -DHPX_WITH_CXX${CXX_STD}=ON"
-configure_extra_options+=" -DHPX_WITH_MALLOC=system"
-configure_extra_options+=" -DHPX_WITH_FETCH_ASIO=ON"
+configure_extra_options+=" -DHPXLocal_WITH_CXX${CXX_STD}=ON"
+configure_extra_options+=" -DHPXLocal_WITH_MALLOC=system"
+configure_extra_options+=" -DHPXLocal_WITH_LOGGING=OFF"
+configure_extra_options+=" -DHPXLocal_WITH_ASYNC_MPI=ON"
```

Comment:
```suggestion
configure_extra_options+=" -DHPXLocal_WITH_MPI=ON"
```
?

Reply:
See: https://github.com/STEllAR-GROUP/hpx-local/pull/2#discussion_r736682825. If there's no preference either way I'll change the variable to `HPXLocal_WITH_MPI`.

Reply:
Why is the MPIAsync stuff part of the local repository now. That doesn't make any sense to me :/

Reply:
Because of a few reasons. Partly because of the current integration with the scheduler (exactly as `async_cuda`). `async_mpi` is neither distributed HPX nor local HPX and I want to separate this out into a separate repository, which I'm pretty sure I've mentioned before. However, that needs a different mechanism for registering the polling functionality, and I don't want to do that at the same time as this split. Since users of the _real_ distributed HPX would have very little reason to use the `async_mpi` module it makes sense to have it here at this point.

Reply:
Note that this means that the only thing distributed HPX needs from local HPX is really the `mpi_base` module, which can eventually stay here, move to distributed HPX or be duplicated in a future `hpx-async-mpi` library.

Reply:
Well, as said - having anything related to MPI in a repository that positions itself as 'local' doesn't sound very sensible to me. The arguments you list (integration with scheduler, registration of polling functions, future plans to separate it, etc.) sound orthogonal to the issue. All of those arguments were equally true and applicable if the MPIAsync module stayed in the distributed part  (where I believe it belongs). 

For the claim that users of distributed HPX have little reason to use MPIAsync - I would beg to disagree. One of our projects (FleCSI) has a diverse set of use cases where we might have to use both, HPX collectives and MPIAsync functionalities at the same time, at least for evaluation purposes.

Reply:
I think my comment about it belonging eventually in a separate repository agrees with your comment that HPXLocal is not the right place long-term. There are technical reasons for having it there now, but my preference is also not to have it there long term.

HPXLocal provides a local-only runtime. That does not mean it can't be used in distributed settings. `async_mpi` is a lightweight wrapper to use asynchronous MPI functions in a runtime that otherwise doesn't know anything about distributed computing.

Finally, you know that we rely on functionality in `async_mpi`. Having it in the distributed HPX means that we will keep depending on the full distributed HPX, which is counter to this whole split. Having it in the local or distributed HPX makes zero difference to FleCSI.

I want to repeat: long-term it doesn't belong in this repository, but it doesn't belong in distributed HPX either. It's an alternative way to do distributed computing in HPX without the distributed runtime.

Reply:
Let's discuss this in person at some point. I'm not having a good feeling about where this is going...

Reply:
Agreed. Let's do that.

Reply:
We discussed this yesterday and I think we're on the same page now. We'll keep `async_mpi` here for the initial split and try to put a more generic mechanism in place in the schedulers to have the MPI polling register itself from a separate library afterwards.

Reply:
Also, regarding the initial question, I'll keep the name since it'll be removed once the functionality is in a separate repository (https://github.com/STEllAR-GROUP/hpx-local/pull/2#discussion_r748137825).